src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
Interesting question!

I like the idea of running annotator on the large code base, like cargo + it's dependencies. I wouldn't worry too much about potentially missing cases, because we are primarily interested in what people actually write. There will be some amount of false positives, bug reports and fixing, but this perfectly OK. And sometimes false positives may actually be on the Rust side, as in the case with naming conventions for static variables. 

I am more worried about the effect on the testing times. Currently, we have similar parsing tests, and they take too much time to my taste. They are valuable of course, but still... I need to figure out how to split our test suite into fast and slow parts. Then, we can run inspections on Cargo as a part of slow tests, and then add any false positives as fast tests.